meson.build

```diff
@@ -4,12 +4,22 @@ project('Hyprland', 'cpp', 'c',
 
 add_global_arguments('-std=c++23', language: 'cpp')
 
+GIT_BRANCH = run_command('gitinfo.sh', 'GIT_BRANCH').stdout().strip()
+GIT_COMMIT_HASH = run_command('gitinfo.sh', 'GIT_COMMIT_HASH').stdout().strip()
+GIT_COMMIT_MESSAGE = run_command('gitinfo.sh', 'GIT_COMMIT_MESSAGE').stdout().strip()
+GIT_DIRTY = run_command('gitinfo.sh', 'GIT_DIRTY').stdout().strip()
```

Comment:
You want the `check: true` kwarg to run_command, because recent versions of Meson emit a deprecation warning if you don't specify some value of `check`.

It's best to use "true", because that would ideally have been the default all along -- if those commands error out for some crazy reason Meson can error out if the information is required.